### PR TITLE
Implement a Request model for storing booking requests

### PR DESCRIPTION
**Pull request by simalicrum**
### What this PR does (required):

Added schema for the request model as follows:

- user_id (user who is requesting)
- sitter_id (pet sitter user id)
- start (date requested to start)
- end (date requested to end)
- accepted (default False)
- declined (default False)
- paid (default False)

### CODE REVIEW

File: server/models/Request.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcryptjs");
+
+const requestSchema = new mongoose.Schema({
+  user_id: {
```
Comment (ugur-akin) [inline server/models/Request.js:5]: Let's use `camelCase` convention for our field names.

File: server/models/Request.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  accepted: {
+    type: Boolean,
+    default: false,
+    required: true,
+  },
+  declined: {
+    type: Boolean,
+    default: false,
+    required: true,
+  },
```
Comment (ugur-akin) [inline server/models/Request.js:32]: I think we can be more flexible and future-proof if we reduce these two into a single field for these two (e.g. `status`).

File: server/models/Request.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcryptjs");
```
Comment (ugur-akin) [inline server/models/Request.js:2]: This doesn't seem to be used, can we remove it?

**Review by ugur-akin — CHANGES_REQUESTED** [review]
Good work! Left a few minor suggestions, let's check back after you have a look and we should be able to merge it soon!

File: server/models/Request.js (hunk 2 above)
Comment (simalicrum) [reply server/models/Request.js]: fixed

**Review by rajivtitus — APPROVED** [review]
Great work! 👍🏽 

Thanks for making the requested changes

**Review by ugur-akin — APPROVED** [review]
LGTM! Good work, the model looks really nice! 💯

**Pull request merged**